kilkari/src/main/java/org/motechproject/nms/kilkari/contract/RchMotherRecord.java

```diff
@@ -281,20 +281,20 @@ public Integer getEntryType() {
     @JsonProperty("Entry_Type")
     public void setEntryType(Object entryType) {
         if (entryType instanceof Integer) {
-            // If the value is already an integer, just assign it directly
             this.entryType = (Integer) entryType;
         } else if (entryType instanceof String) {
-            String entryTypeStr = (String) entryType;
-            if ("ACTIVE".equalsIgnoreCase(entryTypeStr) || entryTypeStr.isEmpty()) {
-                // If the string is "ACTIVE" or empty, assign 1
-                this.entryType = 1;
```

Comment:
To refactor this code using a design pattern, we can use the Strategy Pattern. This pattern allows us to encapsulate different entry type parsing behaviors in separate classes and choose the appropriate one at runtime based on the type of entryType object received. This makes the code more modular, easier to maintain, and extensible if more entry types are added in the future.


Here’s how to refactor the code with the Strategy Pattern:

### Step 1: Define the Strategy Interface
Create an interface for different strategies of parsing entryType.


```
public interface EntryTypeStrategy {
    Integer parseEntryType(Object entryType);
}
```
**Step 2: Implement Concrete Strategy Classes**
Create separate classes for handling Integer, String, and Default types.

**Integer Strategy**

```
public class IntegerEntryTypeStrategy implements EntryTypeStrategy {
    @Override
    public Integer parseEntryType(Object entryType) {
        return (Integer) entryType;
    }
}
```
**String Strategy**


```
public class StringEntryTypeStrategy implements EntryTypeStrategy {
    @Override
    public Integer parseEntryType(Object entryType) {
        String entryTypeStr = ((String) entryType).toUpperCase();
        if ("ACTIVE".equals(entryTypeStr) || "".equals(entryTypeStr)) {
            return 1;
        } else {
            return 9;
        }
    }
}

```
**Default Strategy**

```
public class DefaultEntryTypeStrategy implements EntryTypeStrategy {
    @Override
    public Integer parseEntryType(Object entryType) {
        return 1; // Default value, assuming null or other unsupported type means "ACTIVE"
    }
}
```
### Step 3: Create a Factory for Choosing the Appropriate Strategy
Create a factory class that returns the appropriate strategy based on the type of entryType.


```
public class EntryTypeStrategyFactory {
    public static EntryTypeStrategy getStrategy(Object entryType) {
        if (entryType instanceof Integer) {
            return new IntegerEntryTypeStrategy();
        } else if (entryType instanceof String) {
            return new StringEntryTypeStrategy();
        } else {
            return new DefaultEntryTypeStrategy();
        }
    }
}
```
**Step 4: Refactor the setEntryType Method**
Use the factory to get the appropriate strategy and assign entryType.


```
import com.fasterxml.jackson.annotation.JsonProperty;

public class EntryTypeHandler {

    private Integer entryType;

    @JsonProperty("Entry_Type")
    public void setEntryType(Object entryType) {
        EntryTypeStrategy strategy = EntryTypeStrategyFactory.getStrategy(entryType);
        this.entryType = strategy.parseEntryType(entryType);
    }

    public Integer getEntryType() {
        return entryType;
    }
}

```